### ISSUE TITLE
[BERT/PyTorch] proper way to get an score after inference/predictions and checkpoints/bert_uncased.pt

### ISSUE THREAD

**Issue by gieflij** [NVIDIA/DeepLearningExamples#1181]
Related to **BERT/PyTorhc** 

**Describe the bug**
There's no bug, I just want to know whether I get an inference score in a proper way.

**To Reproduce**
*I followed Quick Start Guide*
1. 'git clone https://github.com/NVIDIA/DeepLearningExamples.git'
2. 'cd DeepLearningExamples/PyTorch/LanguageModeling/BERT'
3. Download the NVIDIA pre-trained checkpoint named 'bert-base-uncased-qa'. (https://catalog.ngc.nvidia.com/orgs/nvidia/models/bert_pyt_ckpt_base_qa_squad11_amp/files)
    - I read the comments that "BERT-Base TensorFlow checkpoint finetuned for QA on SQuAD v1.1".
    - I think if I use this checkpoint, then I don't need to finetuning on it.
    - So I skipped fine-tuning step.
4. Run 'bash scripts/docker/build.sh'
5. Run 'bash scripts/docker/launch.sh'
6. Comments out some codes of /workspace/bert/data/create_datasets_from_start.sh
    - Because my computer stops during wikipedia download
    - I only need vocab.txt and SQuAD
    - Change the codes like : 
***
    #download_wikipedia --outdir ${BERT_PREP_WORKING_DIR}/wikipedia/
    python3 /workspace/bert/data/bertPrep.py --action download --dataset google_pretrained_weights  # Includes vocab
    python3 /workspace/bert/data/bertPrep.py --action download --dataset squad
    #python3 /workspace/bert/data/bertPrep.py --action download --dataset mrpc
    #python3 /workspace/bert/data/bertPrep.py --action download --dataset sst-2
***
 
7. Run '/workspace/bert/data/create_datasets_from_start.sh'
8. Change some codes of /workspace/bert/scripts/run_squad.sh
    - Change the number of GPU from 8 to 1 (for using only one GPU) : num_gpu=${7:-"1"}
    - Change the vocab_file (for using bert base model) : vocab_file=${10:-"$BERT_PREP_WORKING_DIR/download/google_pretrained_weights/uncased_L-12_H-768_A-12/vocab.txt"}
    - Change the mode (for doing inference/predictions) : mode=${11:-"prediction"}
    - Change the config_file (for using bert base model) : CONFIG_FILE=${13:-"/workspace/bert/bert_configs/bert.json"}
 9. Run 'bash scripts/run_squad.sh checkpoints/bert_base_qa.pt'
***
Container nvidia build =  29224839
out dir is /workspace/bert/results/SQuAD
python   run_squad.py --init_checkpoint=/workspace/bert/checkpoints/bert_base_qa.pt --do_predict --predict_file=/workspace/bert/data/squad/v1.1/dev-v1.1.json --predict_batch_size=4  --do_lower_case  --bert_model=bert-base-uncased  --learning_rate=3e-5  --warmup_proportion=0.1 --seed=1  --num_train_epochs=2.0  --max_seq_length=384  --doc_stride=128  --output_dir=/workspace/bert/results/SQuAD  --vocab_file=/workspace/bert/download_from_google-research/uncased_L-12_H-768_A-12/vocab.txt  --config_file=/workspace/bert/bert_configs/base.json  --max_steps=-1   |& tee /workspace/bert/results/SQuAD/logfile.txt
device: cuda n_gpu: 1, distributed training: False, 16-bits training: False
DLL 2022-07-25 06:36:30.739441 - PARAMETER Config : ["Namespace(amp=False, bert_model='bert-base-uncased', cache_dir=None, config_file='/workspace/bert/bert_configs/base.json', disable_progress_bar=False, do_eval=False, do_lower_case=True, do_predict=True, do_train=False, doc_stride=128, eval_script='evaluate.py', fp16=False, gradient_accumulation_steps=1, init_checkpoint='/workspace/bert/checkpoints/bert_base_qa.pt', json_summary='results/dllogger.json', learning_rate=3e-05, local_rank=-1, log_freq=50, loss_scale=0, max_answer_length=30, max_query_length=64, max_seq_length=384, max_steps=-1.0, n_best_size=20, no_cuda=False, null_score_diff_threshold=0.0, num_train_epochs=2.0, output_dir='/workspace/bert/results/SQuAD', predict_batch_size=4, predict_file='/workspace/bert/data/squad/v1.1/dev-v1.1.json', profile=False, seed=1, skip_cache=False, skip_checkpoint=False, train_batch_size=32, train_file=None, use_env=False, verbose_logging=False, version_2_with_negative=False, vocab_file='/workspace/bert/download_from_google-research/uncased_L-12_H-768_A-12/vocab.txt', warmup_proportion=0.1)"] 
DLL 2022-07-25 06:36:30.739831 - PARAMETER SEED : 1 
DLL 2022-07-25 06:36:32.439103 - PARAMETER loading_checkpoint : True 
DLL 2022-07-25 06:36:32.788927 - PARAMETER loaded_checkpoint : True 
DLL 2022-07-25 06:36:34.621834 - PARAMETER model_weights_num : 109488386 
DLL 2022-07-25 06:37:16.983340 - PARAMETER infer_start : True 
DLL 2022-07-25 06:37:16.983407 - PARAMETER eval_samples : 10570 
DLL 2022-07-25 06:37:16.983434 - PARAMETER eval_features : 10833 
DLL 2022-07-25 06:37:16.983458 - PARAMETER predict_batch_size : 4 
DLL 2022-07-25 06:37:17.400299 - PARAMETER eval_start : True 
Evaluating: 100%|██████████| 2709/2709 [00:47<00:00, 57.25it/s]
DLL 2022-07-25 06:37:17.408913 - PARAMETER sample_number : 0 
DLL 2022-07-25 06:37:23.106265 - PARAMETER sample_number : 1000 
DLL 2022-07-25 06:37:27.286693 - PARAMETER sample_number : 2000 
DLL 2022-07-25 06:37:31.477829 - PARAMETER sample_number : 3000 
DLL 2022-07-25 06:37:35.667868 - PARAMETER sample_number : 4000 
DLL 2022-07-25 06:37:39.873206 - PARAMETER sample_number : 5000 
DLL 2022-07-25 06:37:44.088408 - PARAMETER sample_number : 6000 
DLL 2022-07-25 06:37:48.296397 - PARAMETER sample_number : 7000 
DLL 2022-07-25 06:37:52.503509 - PARAMETER sample_number : 8000 
DLL 2022-07-25 06:37:56.949527 - PARAMETER sample_number : 9000 
DLL 2022-07-25 06:38:01.179718 - PARAMETER sample_number : 10000 
DLL 2022-07-25 06:38:58.204230 -  e2e_inference_time : 47.3193244934082  inference_sequences_per_second : 228.93395279784872 

real	2m29.750s
user	2m27.501s
sys	0m2.513s
***

 11. Run 'python data/squad/v1.1/evaluate-v1.1.py data/squad/v1.1/dev-v1.1.json results/SQuAD/predictions.json'
    - I checked /workspace/bert/results/SQuAD/predicitons.json is changed after running 'bash scripts/run_squad.sh checkpoints/bert_base_qa.pt''
    - I got "exact_match" and "f1" score

***
{"exact_match": 81.47587511825922, "f1": 88.68774744997866}
***

**Expected behavior**
1. Is it a proper way for getting an score and is it a reliable score?
3. I cannot find where is checkpoints/bert_uncased.pt. If somebody knows where it is, tell me please.
4. Is it okay use checkpoints/bert_base_qa.pt instead of checkpoints/bert_uncased.pt?

**Environment**
Please provide at least:
* Container version (e.g. pytorch:19.05-py3): bert:latest
* GPUs in the system: (e.g. 8x Tesla V100-SXM2-16GB): 3080TI
* CUDA driver version (e.g. 418.67): 510.73.05

**Comment by sharathts** [comment]
yes, accuracy for bert-base looks as expected.
